core/cloudflow-localrunner/src/main/scala/cloudflow/localrunner/LocalRunner.scala

```diff
@@ -80,13 +82,16 @@ object LocalRunner extends StreamletLoader {
     JRuntime.getRuntime.addShutdownHook(shutdownHook(fos))
 
     Console.withOut(fos) {
-      System.setOut(new PrintStream(fos))
-      readDescriptorFile(appDescriptorFilename) match {
-        case Success(applicationDescriptor) ⇒
-          run(applicationDescriptor, localConfig)
-        case Failure(ex) ⇒
-          log.error(s"Failed JSON unmarshalling of application descriptor file [${appDescriptorFilename}].", ex)
-          System.exit(1)
+      Console.withErr(fos) {
+        System.setOut(new PrintStream(fos))
+        System.setErr(new PrintStream(fos))
```

Comment:
Streams tend to leave resources open if they are not carefully closed, I think that this piece should be enclosed in a `try` block and the Streams closed in the `finally` clause.

Reply:
The fos is closed in shutdown hook, the Printstream closes only what it writes to (which is the fos), so it is not necessary.

Reply:
I am commenting under the assumption that the outermost Stream should be closed (in this case the newly created `PrintStream`).
Not too fussy on this though!